brewtils/rest/system_client.py

```diff
@@ -248,13 +256,16 @@ def send_bg_request(self, **kwargs):
                     return self.send_bg_request(**kwargs)
             raise
 
-        if self._blocking:
-            return self._wait_for_request(request,
-                                          kwargs.get('_raise_on_error', self._raise_on_error))
-        else:
+        # If not blocking just return the future
+        if not blocking:
             return self._thread_pool.submit(self._wait_for_request,
-                                            request,
-                                            kwargs.get('_raise_on_error', self._raise_on_error))
+                                            request, raise_on_error)
+
+        # At this point the request will be complete so check for error
```

Comment:
This breaks backwards compatibility.

The biggest problem I see is if someone is using a newer `brewtils` with an older beer-garden, the `blocking` flag suddenly doesn't work. I think the right thing to do is to introduce some backwards compatibility logic here. I think we should be able to look at the status code for more information (this might come with a change on the API side as well). In any case _something_ returned from the API should tell us the difference between:

* The request has been created, you can query the results here.
* The request was created and successfully executed.

We can then pass the `blocking` flag in the params, if the request is not in bucket 2, then we know we're on an older version of the API and we have to block manually.